jenkins/Jenkinsfile

```diff
@@ -1,374 +1,496 @@
+                                    // Dataproc 1.2 is used for the legacy image and must have python 3.6 installed
+                                    // Otherwise, a terra-docker image is used which must have python 3.7 installed
+                                    def python_version = dp_version.startsWith("1.2") ? "3.6.8" : "3.7.4"
+                                    println("Installing Python $python_version on Dataproc $dp_version...")
+                                    sh(
+                                        script: """
+                                            sed -i 's/python_version=.*/python_version="$python_version"/' jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh
+                                            cat jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh
+                                            cd jenkins/dataproc-custom-images/dataproc-custom-images
+                                            python generate_custom_image.py \
+                                                --image-name "$customDataprocImageBaseName-$dp_version_formatted-$imageID" \
+                                                --dataproc-version "$dp_version" \
+                                                --customization-script ../prepare-custom-leonardo-jupyter-dataproc-image.sh \
+                                                --zone $ZONE \
+                                                --gcs-bucket $DATAPROC_IMAGE_BUCKET \
+                                                --project-id=$GOOGLE_PROJECT \
+                                                --disk-size=50
+                                        """
+                                    )
+                                }
+                            }
+                        }
+                }
+
+                stage('Run GCE image generation script') {
+                    when { expression { GENERATE_GCE_IMAGE == "true" } }
+                        steps {
+                            sh """
+                                docker pull gcr.io/compute-image-tools/daisy:latest
+                            """
+
+                            script {
+                                String daisyImage = "gcr.io/compute-image-tools/daisy:latest"
+                                sh """
+                                    # Create the Daisy scratch bucket if it doesn't exist. The Daisy workflow will clean it up at the end.
+                                    gsutil ls $GCE_IMAGE_BUCKET || gsutil mb -p $GOOGLE_PROJECT -l $REGION $GCE_IMAGE_BUCKET
+
+                                    docker pull $daisyImage
+                                    docker run -i --rm -v ${WORKSPACE}/jenkins/gce-custom-images:/daisy_source_files \
+                                        $daisyImage \
+                                        -project $GOOGLE_PROJECT \
```

Comment:
By the way, might as well mention it here: Chelsea (DevOps) mentioned to me recently that Google has flagged VMs in the `broad-dsp-gcr-public` making legacy calls to the GCE metadata server, which will soon be deprecated. Since the VMs are managed via the [dataproc script](https://cloud.google.com/dataproc/docs/guides/dataproc-images#generating_a_custom_image) (and now Daisy) I'm not sure what we can do about it. I recommended that we wait until the date and see what (if anything) breaks. :)


https://broadworkbench.atlassian.net/browse/DDO-86?focusedCommentId=27791&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-27791

Reply:
👍 Thanks for the heads-up!